### PR TITLE
Bwa gpu

### PR DESCRIPTION
**Pull request by jonca79**
### This PR:

Added: GPU alignment

### Review and tests: 
- [ ] Tests pass
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Code review
- [ ] `CHANGELOG.md` is updated
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

**Comment by Smeds** [comment]
@jonca79  some file contains capital letters and according to our snakemake document are file/folder names only allowed to contain lower cases

File: workflow/rules/bqsr_gpu.smk [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        "alignment/{rule}/{sample}_{type}.recal.txt.log",
+    benchmark:
+        repeat(
+            "alignment/{rule}/{sample}_{type}.recal.txt.benchmark.tsv",
+            config.get("bqsr_gpu", {}).get("benchmark_repeats", 1),
+        )
+    threads: config.get("bqsr_gpu", config["default_resources"])["threads"]
+    # container:
+    #     config.get("align_dedup_GPU", {}).get("container", "default_container")
+    conda:
```
Comment (Smeds) [inline workflow/rules/bqsr_gpu.smk:24]: Should we maybe remove all conda stuff, since it cannot be used?

Comment (jonca79) [reply workflow/rules/bqsr_gpu.smk]: Done

File: workflow/rules/bqsr_gpu.smk [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        recal=temp("alignment/{rule}/{sample}_{type}.recal.txt"),
+    log:
+        "alignment/{rule}/{sample}_{type}.recal.txt.log",
+    benchmark:
+        repeat(
+            "alignment/{rule}/{sample}_{type}.recal.txt.benchmark.tsv",
+            config.get("bqsr_gpu", {}).get("benchmark_repeats", 1),
+        )
+    threads: config.get("bqsr_gpu", config["default_resources"])["threads"]
+    # container:
```
Comment (Smeds) [inline workflow/rules/bqsr_gpu.smk:22]: Why is the container part commented?

File: workflow/rules/apply_bqsr_gpu.smk [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        bam=temp("alignment/{rule}/{sample}_{type}.bqsr.dedup.bam"),
+    log:
+        "alignment/{rule}/{sample}_{type}.bqsr.dedup.log",
+    benchmark:
+        repeat(
+            "alignment/{rule}/{sample}_{type}.bqsr.dedup.benchmark.tsv",
+            config.get("apply_bqsr_gpu", {}).get("benchmark_repeats", 1),
+        )
+    threads: config.get("apply_bqsr_gpu", config["default_resources"])["threads"]
+    # container:
```
Comment (Smeds) [inline workflow/rules/apply_bqsr_gpu.smk:22]: Why is the container part commented?

File: workflow/rules/align_dedup_gpu.smk [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        ID=config.get("align_dedup_gpu", "{sample}").get("ID", "{sample}"),
+    log:
+        "alignment/{rule}/{sample}_{type}.dedup.bam.log",
+    benchmark:
+        repeat(
+            "alignment/{rule}/{sample}_{type}.dedup.bam.benchmark.tsv",
+            config.get("align_dedup_gpu", {}).get("benchmark_repeats", 1),
+        )
+    threads: config.get("align_dedup_gpu", config["default_resources"])["threads"]
+    # container:
```
Comment (Smeds) [inline workflow/rules/align_dedup_gpu.smk:26]: Why is the container part commented?

File: workflow/rules/align_dedup_gpu.smk [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+rule align_dedup_gpu:
+    input:
+        fastq1="prealignment/merged/{sample}_{type}_fastq1.fq.gz",
+        fastq2="prealignment/merged/{sample}_{type}_fastq2.fq.gz",
+        ref=config["reference"]["fasta"],
+    output:
+        bam="alignment/{rule}/{sample}_{type}.dedup.bam",
+    params:
+        SM=config.get("align_dedup_gpu", "{sample}").get("SM", "{sample}"),
+        PL=config.get("align_dedup_gpu", "illumina").get("PL", "illumina"),
```
[the remark this answers is not in the document]
Comment (marrip) [reply workflow/rules/align_dedup_gpu.smk:16]: Sounds right, but I think the info should be in `units.tsv` as it is linked to the `fastq`-files.

[the remark this answers is not in the document]
Comment (Smeds) [reply workflow/rules/align_dedup_gpu.smk:16]: sounds reasonable

**Pull request closed without merging**